src/bundle-manifest.ts

```diff
@@ -186,15 +202,61 @@ export function generateBundles(depsIndex: TransitiveDependenciesMap):
         bundles.find((bundle) => setEquals(entrypoints, bundle.entrypoints));
 
     if (!bundle) {
-      const type = 'html-fragment';
+      const type = getBundleTypeForUrl([...entrypoints][0]);
       bundle = new Bundle(type, entrypoints);
       bundles.push(bundle);
     }
     bundle.files.add(dep);
   }
+
   return bundles;
 }
 
+/**
+ * Instances of `<script type="module">` generate synthetic entrypoints in the
+ * depsIndex and are treated as entrypoints during the initial phase of
+ * `generateBundles`.  Any bundle which provides dependencies to a single
+ * synthetic entrypoint of this type (aka a single entrypoint sub-bundle) are
+ * merged back into the bundle for the HTML containing the script tag.
+ *
+ * For example, the following bundles:
+ *   `[a]->[a], [a>1]->[x], [a>1,a>2]->[y], [a>2]->[z]`
```

Comment:
`a>1` and `a>2` represent script tag entrypoints.  Only `x` and `z` are bundled with `a` because they each server only a single script tag entrypoint.  `y` has to be in a separate bundle so that it is not inlined into bundle `a` in both script tags.

Reply:
I've added that text into the JS description comment below the bundle examples in commit d238cec60780fc211ea29302e99aadeaa3ed3780